internal/handlers/common/update.go

```diff
@@ -746,6 +752,91 @@ func processCurrentDateFieldExpression(doc *types.Document, currentDateVal any)
 	return changed, nil
 }
 
+func processBitFieldExpression(command string, doc *types.Document, updateV any) (bool, error) {
+	var changed bool
+
+	bitDoc := updateV.(*types.Document)
+
+	for _, bitKey := range bitDoc.Keys() {
+		bitValue := must.NotFail(bitDoc.Get(bitKey))
+
+		nestedDoc, ok := bitValue.(*types.Document)
+		if !ok {
+			return false, newUpdateError(
+				commonerrors.ErrTypeMismatch,
+				fmt.Sprintf(
+					`The $bit modifier is incompatible with a %s. `+
+						`You must pass in an embedded document: {$bit: {field: {and/or/xor: #} }}`,
+					commonparams.AliasFromType(bitValue),
+				),
+				command,
+			)
+		}
+
+		path := must.NotFail(types.NewPathFromString(bitKey))
+
+		// $bit ignores fields that does not exist
+		if !doc.HasByPath(path) {
+			continue
+		}
+
+		docValue, err := doc.GetByPath(path)
+		if err != nil {
+			return false, err
+		}
+
+		// $bit operations can only be performed on integer fields
+		switch docValue.(type) {
+		case int32, int64:
+		default:
+			return false, newUpdateError(
+				commonerrors.ErrTypeMismatch,
+				fmt.Sprintf(
+					`Cannot apply $bit to a value of non-integral type. `+
+						`{_id: %s} has the field '%s' of non-numeric type %s`,
+					types.FormatAnyValue(must.NotFail(doc.Get("_id"))),
+					bitKey,
+					commonparams.AliasFromType(docValue),
+				),
+				command,
+			)
+		}
+
+		for _, bitOp := range nestedDoc.Keys() {
+			bitOpValue := must.NotFail(nestedDoc.Get(bitOp))
+			fieldType := commonparams.AliasFromType(bitOpValue)
+
+			// $bit operator argument must be an integer (Int32/Long)
+			if fieldType != "int" && fieldType != "long" {
```

Comment:
We are checking int and long type here and in `performBitLogic()`? I don't think we need to check it here.

Instead, let's use the error from `performBitLogic()` and return `newUpdateError` based on that. :hugs: